tests/unit/test_slide.py

```diff
@@ -104,28 +105,51 @@ def it_knows_its_name(self, slide_path, expected_value):
 
         assert name == expected_value
 
-    def it_raises_error_with_unknown_mpp_without_largeimage(self, tmpdir):
-        slide, _ = base_test_slide(tmpdir, PILIMG.RGBA_COLOR_500X500_155_249_240)
+    @pytest.mark.parametrize(
+        "use_largeimage, properties, metadata, expected_value",
+        (
+            (True, {}, {"mm_x": 1}, 1000.0),
+            (False, {"openslide.mpp-x": 33}, None, 33.0),
+            (False, {"aperio.MPP": 33}, None, 33.0),
+            (
+                False,
+                {"tiff.XResolution": 1000, "tiff.ResolutionUnit": "centimeter"},
+                None,
+                10.0,
+            ),
+        ),
+    )
+    def it_knows_its_base_mpp(
+        self, request, use_largeimage, properties, metadata, expected_value
+    ):
+        slide = Slide("foo", "bar", use_largeimage=use_largeimage)
+        property_mock(request, Slide, "properties", return_value=properties)
+        property_mock(request, Slide, "_metadata", return_value=metadata)
+
+        assert slide.base_mpp == expected_value
+
+    def but_it_raises_large_image_error_with_unknown_mpp_without_largeimage(
+        self, request
+    ):
+        slide = Slide("foo", "bar", use_largeimage=False)
+        property_mock(request, Slide, "properties", return_value={})
 
         with pytest.raises(MayNeedLargeImageError) as err:
             slide.base_mpp
 
-        assert isinstance(err.value, MayNeedLargeImageError)
```

Comment:
why removing this?

Reply:
if we use a context manager that check for an exception it's already tested that we are raising that kind of exception. This assert is unnecessary